pandas/core/groupby/generic.py

```diff
@@ -440,13 +446,25 @@ def _get_index() -> Index:
             return result
 
         if isinstance(values[0], Series):
-            return self._concat_objects(keys, values, not_indexed_same=not_indexed_same)
+            return self._concat_objects(
+                keys,
+                values,
+                not_indexed_same=not_indexed_same,
+                override_group_keys=override_group_keys,
+            )
         elif isinstance(values[0], DataFrame):
             # possible that Series -> DataFrame by applied function
-            return self._concat_objects(keys, values, not_indexed_same=not_indexed_same)
+            return self._concat_objects(
+                keys,
+                values,
+                not_indexed_same=not_indexed_same,
+                override_group_keys=override_group_keys,
+            )
         else:
             # GH #6265 #24880
-            result = self.obj._constructor(
+            # ignore Incompatible types in assignment (expression has type
+            #   "Series", variable has type "DataFrame")
+            result = self.obj._constructor(  # type: ignore
```

Comment:
mypy is inferring `result` to be a DataFrame, probably off of the first occurrence of it within this function. Is that inference correct? I think would be better to explicitly type out what `result` can contain than ignoring